src/Core/Search/Discussion/Gambit/AuthorGambit.php

```diff
@@ -50,7 +51,7 @@ protected function conditions(AbstractSearch $search, array $matches, $negate)
 
         $ids = [];
         foreach ($usernames as $username) {
-            $ids[] = $this->users->getIdForUsername($username);
+            $ids[] = is_numeric($username) ? $username : $this->users->getIdForUsername($username);
```

Comment:
I don't think I'd look for a user's posts by their ID. If I wanted to know their username, I'd visit their profile page, `/u/ID`. But some people would prefer to have to do just one query, and not visit another page.
Talking about the profile page, as it allows user IDs, wouldn't it make sense to maybe use that method as well for the gambit? I haven't seen anyone bring this up, and I haven't been able to find where in Flarum's code this happens.